### ISSUE TITLE
collections/<name>/documents/export behavior changed in version 0.24.0

### ISSUE THREAD

**Issue by Bug-Reaper** [typesense/typesense#880]
## Description

Previously we were able to download an entire `ljson` export of a collection from the endpoint via something like
```
curl -H 'X-TYPESENSE-API-KEY: KEY' -X GET "http://<ip>:<port>/collections/<name>/documents/export" 
```
After upgrade to v0.24.0 it seems like this endpoint times out after around one and half minutes and abruptly stops sending data mid response.

Is there some type of new header/request parameter we need to send now to prevent a timeout with the export endpoint?

## Expected Behavior

 We're trying to export a very large collection ~2.3m records

## Actual Behavior

The call abruptly ends after around a minute and a half with only 1gb data sent (approx 125k records). The incomplete file ends abruptly in the middle of a document record. Behavior similar for both `wget` and `curl` attempts.

----

**Typesense Version**:
0.24.0

**OS**:
Arch Linux

**Comment by kishorenc** [comment]
@Bug-Reaper 

We made a change to speed up exports in 0.24 -- they should be much faster. I'm not able to reproduce this issue though: I tried pulling a large 8GB+ collection from an instance South Asia to US west coast, and that export call completed fully without any issues. I used `curl` as well for this. The export ran for 10+ minutes and finished (this can be done faster via larger export batch size, but I used the default). 

Can you try calling curl with `--http2` or `--http1.1` flags to use if it makes a difference?

**Comment by kishorenc** [comment]
@Bug-Reaper 

I think the perf work increased memory usage for exports so that's probably an issue here (the test machine had enough RAM but it might not be the case on yours). I'm going to fix that and post an update here.

**Comment by kishorenc** [comment]
@Bug-Reaper 

I've a fix: how do you deploy Typesense? I can share a build and you can test it out.

**Comment by Bug-Reaper** [comment]
So we deploy on a single instance setup on a VM (Arch Linux) and it's fairly beefy on the memory side of things 64gib.

Found a work-around and so haven't looked into it too much more. 
1. We were able to run wget/curl while ssh-ed into the machine that runs typesense. (this ran longer than a minute and finished properly)
2. scp the generated ljson file from that remote machine to our desired local setup.

I'm not 100% sure why wget/curl work when you run them directly from the typesense host machine. I think this rules out a memory bottleneck though.

Can try curl with the various http flags to see if that makes a difference. Gonna plan to report back with those results as soon as I can.

----

I did notice the speed of the ljson export is massively improved on the latest version. Much appreciated and fantastic work, it felt like a 10x speed jump.
